### PR TITLE
Fix tol for test_nvfuser_correctness__softmax_backward_data_cuda

### PR DESCRIPTION
**Pull request by SherlockNoMad**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #89178

cc @jeffdaily @sunway513 @jithunnair-amd @pruthvistony @ROCmSupport @kevinstephano @jjsjann123

### CODE REVIEW

**Comment by kit1980** [comment]
To give some context on this PR: after https://github.com/pytorch/pytorch/pull/88761, `test_nvfuser_correctness__softmax_backward_data_cuda_bfloat16` and `test_nvfuser_correctness__softmax_backward_data_cuda_float16 ` are failing on rocm: https://github.com/pytorch/pytorch/actions/runs/3483501734/jobs/5827293170#logs

**Comment by ezyang** [comment]
Well, in sherlock's defense, it is half precision. We might need to do the fp64 trick cc @Chillee

File: torch/testing/_internal/common_methods_invocations.py [hunk 1]
```diff
@@ -10596,6 +10596,9 @@ def reference_flatten(input, start_dim=0, end_dim=-1):
         skips=(
             DecorateInfo(unittest.expectedFailure, 'TestCommon', 'test_noncontiguous_samples', device_type='cpu'),
             DecorateInfo(unittest.expectedFailure, 'TestJit', 'test_variant_consistency_jit', dtypes=(torch.float32,)),
+            DecorateInfo(toleranceOverride({torch.float16: tol(atol=0.05, rtol=0.003),
```
Comment (mruberry) [inline torch/testing/_internal/common_methods_invocations.py:10599]: Are the current failures only on ROCm, as @kit1980 suggests? If so maybe these tolerances should only be applied for ROCm?

Comment (kit1980) [reply torch/testing/_internal/common_methods_invocations.py:10599]: > The failing test is "test_nvfuser_correctness". Are we running nvfuser on Rocm GPU? Does we expected this to work? @kit1980 Or shall we `@skipIfRocm` for this test?

I see now that `test_nvfuser_correctness__softmax_backward_data_cuda_bfloat16` and `test_nvfuser_correctness__softmax_backward_data_cuda_float16` were introduced by https://github.com/pytorch/pytorch/pull/8876 and were never run before that.

So yes, we should just disable the tests for ROCm.

**Comment by SherlockNoMad** [comment]
I updated the test data to better reflect the output property of softmax, (summing up to 1), and I can use a more reasonable tol now.

**Review by kit1980 — APPROVED** [review]

**Comment by kit1980** [comment]
> I updated the test data to better reflect the output property of softmax, (summing up to 1), and I can use a more reasonable tol now.

Looks good to me as is to make the CI green, we can try to do something better in the future if needed.

**Comment by ngimel** [comment]
Yes AMD enabled nvfuser, so we are running nvfuser tests on ROCm.
